### ISSUE TITLE
Transcoding DVD hangs on futex_wait_queue_me

### ISSUE THREAD

**Issue by karlrelton** [danielgtaylor/arista#10]
Trying to transcode a DVD disc. Add an item to the queue, and it starts transcoding.
Verbose output reports the gstreamer pipeline, then outputs the regular libdvdcss key stuff, but then it hangs. System monitor shows process is sleeping, waiting on futex_wait_queue_me.

This is in Ubuntu 10.04, but with arista upgraded to 0.9.4.1 and commit 13b05... applied.

I am happy to try and help debugging if you can provide me some directions of what to try next.

**Comment by karlrelton** [comment]
Interestingly depending on the preset I get a different result. E.g. choosing computer-theora it completes in a couple of seconds, giving a zero-second length output file.

The preset that gives a hang is Android-nexus one.

**Comment by danielgtaylor** [comment]
Can you try the latest trunk as of last night? I've made a few changes after chatting with the GStreamer developers to try and alleviate some of the pipe linking issues and other problems.

I tried a DVD yesterday and was able to reproduce this, and with some formats it does indeed seem to work okay. I'll try again later with my latest changes and if it's still an issue there must be some way to fix it, otherwise I'll be opening a few upstream bug reports.

**Comment by karlrelton** [comment]
I've applied the latest patches - and it is now ripping instead of hanging.

That leads me to a new problem though - this DVD disc is a film that starts by first showing an animated menu page that just loops round. Arista is now looping round (i.e. transcoding it, and then transcoding it, and then ...).

I guess I should open a separate ticket for that?

**Comment by danielgtaylor** [comment]
I'm working on a fix for that, trying to figure out all the nuances of DVD handling and title selection with GStreamer. Let's just leave this ticket open for now.

**Comment by danielgtaylor** [comment]
The DVD title selection should be back to how it was before as of 674cc06e589b42995a78090031ef50419840acc9. It seems that rsndvdbin which uridecodebin automagically uses is really only useful for playback and mouse input, and fairly useless for transcoding so I've switched back to using dvdreadsrc and hoping that the first title is the one people want to rip. In the future I hope to add real title selection, but I'm struggling with how to do it.

Please pull my updates and see if you can rip your DVD now.

**Comment by danielgtaylor** [comment]
As of 08cb426a07db9ca429d6bf8e986900c0c78edf43 the most likely title to be the main feature is automatically chosen when ripping from a DVD (instead of it being hardcoded to title 1 like it always has been). This should make DVDs where the main feature isn't the first title rippable. You can always manually specify it on the commandline as well using dvd://path@title where path is the path to the DVD device, e.g. /dev/dvd or /dev/sr0.

**Comment by karlrelton** [comment]
I've been away this week. I'll see if I can pull the patches and test at my end over the weekend.

**Comment by karlrelton** [comment]
Yep - on my system it got to work ripping the longest track. Many thanks.
